src/mon/OSDMonitor.cc

```diff
@@ -4837,6 +4839,34 @@ void OSDMonitor::check_pg_creates_sub(Subscription *sub)
   }
 }
 
+void OSDMonitor::check_for_filestore_osds(health_check_map_t *checks)
+{
+  set<int32_t> filestore_osds;
+  for (int i = 0; i < (int)osdmap.get_num_osds(); ++i) {
+    map<string,string> meta;
+    load_metadata(i, meta, nullptr);
+    auto p = meta.find("osd_objectstore");
+    if (p == meta.end()) continue;
+
+    if (p->second == "filestore") {
+      filestore_osds.insert(i);
+    }
```

Comment:
This can be further simplified using `get_osd_objectstore_type` 

     string objectstore_type;
      int r = get_osd_objectstore_type(i, &objectstore_type);
      if (r < 0 || objectstore_type == "bluestore") {
        filestore_osds.insert(i);
      }